public/app/plugins/panel/datagrid/state.ts

```diff
@@ -219,14 +219,12 @@ export const datagridReducer = (state: DatagridState, action: DatagridAction): D
 
       columns = [
         ...updateColumnsPayload.frame.fields.map((field: Field, index: number) => {
-          const displayName = getFieldDisplayName(field, updateColumnsPayload.frame);
-
           // find column by field name and update width in new set. We cannot use index because
           // if a column gets deleted we don't know the correct index anymore
-          const width = state.columns.find((column) => column.title === displayName)?.width;
+          const width = state.columns.find((column) => column.title === field.name)?.width;
 
           return {
-            title: displayName,
+            title: field.name,
```

Comment:
IIUC, this now ignores displayName/displayNameFromDS -- that seems wrong.  I assume we want to have the same names we show elsewhere.

Can we get the name from getFieldDisplayName(), and only clear the configs when we change the name?

Reply:
You're right, I pushed a fix